stackslib/src/chainstate/stacks/db/transactions.rs

```diff
@@ -2202,9 +2202,9 @@ pub mod test {
         let contract_names = ["hello-world-0", "hello-world-1"];
         let expected_errors = [
             "Tried to close list which isn't open.",
-            ":2:14: invalid variable definition",
+            ":0:0: invalid variable definition",
         ];
-        let expected_errors_2_1 = ["unexpected ')'", ":2:14: invalid variable definition"];
+        let expected_errors_2_1 = ["unexpected ')'", ":0:0: invalid variable definition"];
```

Comment:
I think this should switch on the `developer-mode`:

```rust
    let expected_line_num_error = if cfg!(feature = "developer-mode") {
         ":2:14: invalid variable definition"
    } else {
         ":0:0: invalid variable definition"
    };
    
    let expected_errors = [
            "Tried to close list which isn't open.",
            expected_line_num_error
    ];
```

Reply:
Oh, and also, I was only running the clarity tests in both `developer-mode` and regular mode, not the stackslib tests.

The stackslib tests take a good bit longer -- the clarity tests finish in <4s on my machine, and the stackslib tests all but 2 finish in about 4 minutes, then those 2 run for a while, so I won't update the CI to run these in developer-mode, but I'll still include the change in the test.

Reply:
46b7a81b4